src/core/task.ts

```diff
@@ -20,7 +19,7 @@ export async function schedule(task: TaskFunction): Promise<void> {
         queue.push({task, resolve});
     });
 
-    if (active === null) { run(); }
+    if (active === null) { requestAnimationFrame(run); }
```

Comment:
>requestAnimationFrame [](start = 27, length = 21)

requestIdleCallback would be better? If not add a comment explaining why the requestAnimationFrame, which could block the UI thread with higher priority.

Reply:
There are similar concepts and agreed requestIdleCallback would be ideal but Edge doesn't support requestIdleCallback yet: https://developer.mozilla.org/en-US/docs/Web/API/Window/requestIdleCallback. Instead of getting into pollyfills - I stuck with a stable parallel API. As far as I understand, requestAnimationFrame is essentially gives us an opportunity to execute at the next time UI thread is free. It wouldn't block the thread. It is also consistent with what we were using in wait function in the same file.

Reply:
I don't think you need polyfills. Just encapsulate it like you did setTimer and then you can change it later.

---
In reply to: [344512522](https://github.com/microsoft/clarity-js/pull/158#discussion_r344512522) [](ancestors = 344512522)

Reply:
Right, but requestIdelCallback gets a calback when the browser is "idle", which is not the same as for requestAnimationCallback (which is supposed to be tied to the frame rate of the rendering so that you get one call per frame. 

---
In reply to: [344513493](https://github.com/microsoft/clarity-js/pull/158#discussion_r344513493) [](ancestors = 344513493)